cscs-checks/apps/cp2k/cp2k_check.py

```diff
@@ -13,11 +13,15 @@ def __init__(self, check_name, check_descr, **kwargs):
         self.executable = 'cp2k.psmp'
         self.executable_opts = ['H2O-256.inp']
 
+        energy = sn.extractsingle(r'\s+ENERGY\| Total FORCE_EVAL \( QS \) '
+                                  r'energy \(a\.u\.\):\s+(?P<energy>\S+)',
+                                  self.stdout, 'energy', float, item=-1)
         self.sanity_patterns = sn.all([
             sn.assert_found(r'PROGRAM STOPPED IN', self.stdout),
             sn.assert_eq(sn.count(sn.extractall(
                 r'(?P<step_count>STEP NUM)',
-                self.stdout, 'step_count')), 10)
+                self.stdout, 'step_count')), 10),
+            sn.assert_bounded(energy, -4404.2324, -4404.2322)
```

Comment:
`assert_bounded` is better than `assert_reference` but not as descriptive as we want. Usually, when you check a numerical value, you need the reference value and a sufficiently small value (epsilon) for the acceptable deviation from the reference. The `assert_bounded` doesn't show those two. Here is what I propose:

```python
energy  = sn.extractsingle(...)
energy_diff = sn.abs(energy - REFERENCE)  # Put the reference energy value here
self.sanity_patterns = sn.all([
    ...
    sn.assert_lt(energy_diff, 1e-6)    # put a reasonable epsilon value is for this case.
])
```